### ISSUE TITLE
[TearsheetNarrow]: Keyboard focus doesn't work on stacked tearsheet

### ISSUE THREAD

**Issue by tm-chi** [carbon-design-system/ibm-products#5829]
### Package

Carbon for IBM Products

### Description

When tearsheet stacks on another tearsheet, keyboard focus moves to the stacked tearsheet, but tab key doesn't work. 

### Component(s) impacted

Tearsheet, TearsheetNarrow

### Browser

Chrome

### @carbon/ibm-products (previously @carbon/ibm-cloud-cognitive) version

2.46.0

### Suggested Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Product/offering

ibm-products

### CodeSandbox or Stackblitz example

https://stackblitz.com/edit/github-gff4kw?file=src%2FExample%2FExample.jsx

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues

**Comment by tm-chi** [comment]
I heard that keyboard focus doesn't work when tearsheets is in parent child hierarchy. The stacked tearsheets should be siblings.
I close the issue.
